File: frontend/src/types/block-type.ts
export const BlockType = {
    Paragraph: "Paragraph",
    Heading1: "Heading1",
    Heading2: "Heading2",
    Heading3: "Heading3",
    BulletList: "BulletList",
    OrderedList: "OrderedList",
    TaskItem: "TaskItem",
    Image: "Image",
    File: "File",
    Video: "Video",
    CodeBlock: "CodeBlock",
    Divider: "Divider",
    Quote: "Quote"
} as const;

export type BlockType = typeof BlockType[keyof typeof BlockType];

File: server/Domain/Entities/Document/BlockType.cs
using System.Text.Json.Serialization;

namespace Domain.Entities;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum BlockType 
{
    Paragraph,
    Heading1,
    Heading2,
    Heading3,
    BulletList,
    OrderedList,
    TaskItem,
    Image,
    File,
    Video,
    CodeBlock,
    Divider,
    Quote
}
